h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -1077,14 +1077,31 @@ public String getPlanSQL() {
         StatementBuilder buff = new StatementBuilder();
         for (TableFilter f : topFilters) {
             Table t = f.getTable();
+            boolean isPersistent = !t.isTemporary();
             if (t.isView() && ((TableView) t).isRecursive()) {
-                buff.append("WITH RECURSIVE ").append(t.getName()).append('(');
+                buff.append("WITH RECURSIVE ");
+                if(isPersistent){
+                    buff.append("PERSISTENT ");
+                }
+                buff.append(t.getName()).append('(');
                 buff.resetCount();
                 for (Column c : t.getColumns()) {
                     buff.appendExceptFirst(",");
                     buff.append(c.getName());
                 }
-                buff.append(") AS ").append(t.getSQL()).append("\n");
+                String theSQL = t.getSQL();
+                if(!(theSQL.startsWith("(")&&theSQL.endsWith(")"))){
```

Comment:
This code was added because recursive SQL from plan was trying to generate

WITH cte(col1,col2) AS public.cte_underlying_table
instead of:
WITH cte(col1,col2) AS  (SELECT col1,col2 FROM public.cte_underlying_table)